python/ray/data/tests/test_zip.py

```diff
@@ -82,34 +82,50 @@ def test_zip_pandas(ray_start_regular_shared):
     ds2 = ray.data.from_pandas(pd.DataFrame({"col3": ["a", "b"], "col4": ["d", "e"]}))
     ds = ds1.zip(ds2)
     assert ds.count() == 2
-    assert "{col1: int64, col2: int64, col3: object, col4: object}" in str(ds)
+
     result = list(ds.take())
     assert result[0] == {"col1": 1, "col2": 4, "col3": "a", "col4": "d"}
 
+    # Execute the dataset to get full schema.
+    ds = ds.materialize()
+    assert "{col1: int64, col2: int64, col3: object, col4: object}" in str(ds)
+
```

Comment:
Need to update tests for `Zip`, because in the test, we call `ds.count()` before attempting to check the schema from the `Dataset.__str__` representation. After updating `ds.count()` to no longer execute and get the list of underlying Blocks, the schema is unknown for N-ary operators without executing: https://github.com/ray-project/ray/blob/master/python/ray/data/_internal/plan.py#L155-L158